src/core/registry/server.go

```
@@ -95,10 +100,17 @@ func NewServer(entDB *database.EntDatabase, config *RegistryConfig, logger *logg
-	// Initialize trust chain if TLS is not "off"
+	// Initialize trust chain if TLS is not "off". A failure here is fatal:
+	// see initTrustChain doc and issue #989 — silently dropping a backend
+	// would leave the registry running but rejecting every heartbeat with
+	// "no backends configured", which is exactly the bug we're closing.
 	var trustChain *trust.TrustChain
 	if config.TlsMode != "" && config.TlsMode != "off" {
-		trustChain = initTrustChain(config, logger)
+		var err error
+		trustChain, err = initTrustChain(config, logger)
+		if err != nil {
+			return nil, fmt.Errorf("trust chain init: %w", err)
```

golangci-lint (2.12.2)

[error] lostcancel: this return statement may be reached without using the shutdownCancel var defined on line 77

(govet)